### PR TITLE
(0014626) Add data source aviatrix_firewall

### PR DESCRIPTION
**Pull request by CyrusJavan**
- Add data source `aviatrix_firewall` for pulling data about a stateful firewall
- Add tests and documentation for the data source

### CODE REVIEW

File: aviatrix/data_source_aviatrix_firewall.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aviatrix/goaviatrix"
+	"log"
+)
+
+func dataSourceAviatrixFirewall() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAviatrixFirewallRead,
```
Comment (arvindsree) [inline aviatrix/data_source_aviatrix_firewall.go:11]: Is it possible to re-use ResourceAviatrixFirewallRead?

Comment (arvindsree) [reply aviatrix/data_source_aviatrix_firewall.go:11]: I guess not..the import works differently

**Review by arvindsree — APPROVED** [review]

**Comment by arvindsree** [comment]
Update the description to summarize the changes

**Comment by arvindsree** [comment]
Update PR titles to contain the mantis id and use the "Squash and merge" option while merging

**Review by arvindsree — APPROVED** [review]

File: aviatrix/data_source_aviatrix_firewall_test.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	account_name = aviatrix_account.test.account_name
+	name         = "tfv-%s"
+	region       = "%s"
+	cidr         = "10.0.0.0/16"
+}
+data "aviatrix_vpc" "test" {
+	name = aviatrix_vpc.test.name
+}
+resource "aviatrix_gateway" "test" {
+  cloud_type   = 1
```
Comment (zhiqiangjin) [inline aviatrix/data_source_aviatrix_firewall_test.go:64]: Need to keep same indent format.

File: aviatrix/data_source_aviatrix_firewall_test.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  account_name = aviatrix_account.test.account_name
+  gw_name      = "test-gw-%s"
+  vpc_id       = aviatrix_vpc.test.vpc_id
+  vpc_reg      = "%s"
+  gw_size      = "t2.micro"
+  subnet       = data.aviatrix_vpc.test.public_subnets[0].cidr
```
Comment (zhiqiangjin) [inline aviatrix/data_source_aviatrix_firewall_test.go]: Need to keep same indent format.

File: aviatrix/data_source_aviatrix_firewall_test.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  base_log_enabled = true
+
+  policy {
+    protocol    = "tcp"
+    src_ip      = "10.15.0.224/32"
+    log_enabled = false
+    dst_ip      = "10.12.0.172/32"
+    action      = "allow"
+    port        = "0:65535"
+    description = "This is policy no.1"
```
Comment (zhiqiangjin) [inline aviatrix/data_source_aviatrix_firewall_test.go]: Need to keep same indent format.

File: aviatrix/data_source_aviatrix_firewall_test.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+  policy {
+    protocol    = "tcp"
+    src_ip      = "10.15.1.224/32"
+    log_enabled = true
+    dst_ip      = "10.12.1.172/32"
+    action      = "deny"
+    port        = "0:65535"
+    description = "This is policy no.2"
+  }
```
Comment (zhiqiangjin) [inline aviatrix/data_source_aviatrix_firewall_test.go]: Need to keep same indent format.

File: aviatrix/data_source_aviatrix_firewall_test.go [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+}
+data "aviatrix_firewall" "test" {
+  gw_name = aviatrix_firewall.test.gw_name
```
Comment (zhiqiangjin) [inline aviatrix/data_source_aviatrix_firewall_test.go]: Need to keep same indent format.

File: website/docs/d/aviatrix_firewall.html.markdown [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+# aviatrix_firewall
+
+Use this data source to get the Aviatrix stateful firewall for use in other resources.
+
+## Example Usage
+
+```hcl
+# Aviatrix Firewall Data Source
+data "aviatrix_firewall" "foo" {
+	gw_name = "gw-abcd"
```
Comment (zhiqiangjin) [inline website/docs/d/aviatrix_firewall.html.markdown]: Need to keep same indent format.

File: aviatrix/data_source_aviatrix_firewall.go [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+		policies = append(policies, policyToMap(p))
+	}
+	d.Set("policies", policies)
+
+	d.SetId(gwName)
+
+	return nil
+}
+
+func policyToMap(p *goaviatrix.Policy) map[string]interface{} {
```
Comment (zhiqiangjin) [inline aviatrix/data_source_aviatrix_firewall.go]: If this func can also be used inside resource_aviatrix_firewall, better move it to firewall.go

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**